### PR TITLE
FEAT: Add Video-listing and data fetching from backend

### PR DESCRIPTION
**Pull request by VishalPatil18**
### 📝 Description

Add Video-listing and data fetching from backend. Add Mock database to display it on videos page.

### 🚀 New behavior

- The data is now dynamic and is loaded from the server using the `get` request on the mock backend server.

### 🖼 Preview Link

[https://vistream-react-version-o3vah76lw-vishalpatil18.vercel.app/](https://vistream-react-version-o3vah76lw-vishalpatil18.vercel.app/)

### 💣 Is this a breaking change?
- [ ] Yes
- [x] No

### 🔬 Is the code self-reviewed?
- [x] Yes
- [ ] No

### 📋 Is the code well-formatted?
- [x] Yes
- [ ] No

### 🔗 Related to Issue
- #8 

> NOTE: The site is not responsive yet. Will work on responsiveness at the end.

### CODE REVIEW

**Review by AnkurChunekar — APPROVED** [review]
Code is very neat and clean, good job @VishalPatil18

File: src/components/FeaturedVideos/FeaturedVideos.jsx [hunk 1]
```diff
@@ -30,7 +29,7 @@ const FeaturedVideos = ({ page = "" }) => {
         ) : null}
       </div>
       <div className={styles.featuredVideos}>
-        {videoList.map((video, i) => (
+        {videos.map((video, i) => (
           <MainVideoCard key={i} video={video} />
```
Comment (kart-c) [inline src/components/FeaturedVideos/FeaturedVideos.jsx:33]: since your backend database is complete, you can use `video._id` as your key here

**Review by kart-c — APPROVED** [review]
Change all the `key` attributes to use `_id` of your product instead of `index` now that you have a database. Other than that, your code is really clean

File: src/components/FeaturedVideos/FeaturedVideos.jsx (hunk 1 above)
Comment (VishalPatil18) [reply src/components/FeaturedVideos/FeaturedVideos.jsx:33]: Will correct this in the next PR. Thanks for pointing it out!

**Pull request merged**